Description: Test if  environment variable "HOME" exist
Author: Jörg Frings-Fürst <debian@jff-webhosting.net>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=786452
Forwarded: https://alioth.debian.org/tracker/index.php?func=detail&aid=315071&group_id=30186&atid=410366
Last-Update: 2015-05-22
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: trunk/backend/artec_eplus48u.c
===================================================================
--- trunk.orig/backend/artec_eplus48u.c
+++ trunk/backend/artec_eplus48u.c
@@ -3580,6 +3580,14 @@ load_calibration_data (Artec48U_Scanner
 
   s->calibrated = SANE_FALSE;
   path[0] = 0;
+
+  /* return SANE_STATUS_INVAL if HOME environment variable is not set */
+  if (getenv ("HOME") == NULL)
+  {
+    XDBG ((1, "Environment variable HOME not set\n"));
+    return SANE_STATUS_INVAL;
+  }
+
   if (strlen (getenv ("HOME")) < (PATH_MAX - 1))
     strcat (path, getenv ("HOME"));
   else
@@ -3690,6 +3698,14 @@ save_calibration_data (Artec48U_Scanner
   mode_t mode = S_IRUSR | S_IWUSR;
 
   path[0] = 0;
+
+  /* return SANE_STATUS_INVAL if HOME environment variable is not set */
+  if (getenv ("HOME") == NULL)
+  {
+    XDBG ((1, "Environment variable HOME not set\n"));
+    return SANE_STATUS_INVAL;
+  }
+
   if (strlen (getenv ("HOME")) < (PATH_MAX - 1))
     strcat (path, getenv ("HOME"));
   else
